src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -31,11 +32,18 @@ public BitbucketPluginConfiguration() {
     @Override
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
         req.bindJSON(this, json);
-        if (getServerList().stream().allMatch(server -> server.validate())) {
+        List<FormValidation> aggregate = new ArrayList<>();
+        getServerList().stream().map(BitbucketServerConfiguration::validate).forEach(aggregate::addAll);
```

Comment:
If you remove the `List<FormValidation> aggregate` declaration above you can then change this line to
`List<FormValidation> aggregate = getServerList().stream().map(BitbucketServerConfiguration::validate).flatMap(Collection::stream)
                .filter(validation -> validation.kind != Kind.OK).collect(Collectors.toList());`
And you have a (long) one-liner that does all the things.

Reply:
Glad to hear there was a better way! Patched in.